libraries/AP_UAVCAN/AP_UAVCAN_DNA_Server.cpp

```diff
@@ -541,7 +547,7 @@ void AP_UAVCAN_DNA_Server::handleNodeInfo(uint8_t node_id, uint8_t unique_id[],
                 nodeInfo_resp_rcvd = true;
             }
             setVerificationMask(node_id);
-        } else {
+        } else if (!_ap_uavcan->check_option(AP_UAVCAN::Options::DNA_IGNORE_DUPLICATE_NODE)) {
```

Comment:
I believe a better, safer way is to allow replacement, not duplication. Maybe something like if a node pops up with same node id replace it, and then if a node shows up again with same node ID within X amount of time, raise alarm.

This might not be required for tightly controlled setups. But for DIY and prototypes, there should at least be some sort of disclaimer put on this, so people don't misuse this option.